pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,213 @@
-			return restart(ctx, targetCluster, args...)
+			return restart(ctx, args...)
 		},
 	}
-	command.Flags().StringVarP(&targetCluster, "target-cluster", "t", "", "The target cluster")
-	flags.MustMarkRequired(command, "target-cluster")
 	return command
 }
 
-func restart(ctx *clicontext.CommandContext, clusterName string, deployments ...string) error {
+func restart(ctx *clicontext.CommandContext, clusterNames ...string) error {
+	clusterName := clusterNames[0]
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
+	ioStreams := genericclioptions.IOStreams{
+		In:     os.Stdin,
+		Out:    os.Stdout,
+		ErrOut: os.Stderr,
+	}
+	kubeConfigFlags.ClusterName = nil  // `cluster` flag is redefined for our own purpose
+	kubeConfigFlags.AuthInfoName = nil // unused here, so we can hide it
+	kubeConfigFlags.Context = nil      // unused here, so we can hide it
+
 	cfg, err := configuration.LoadClusterConfig(ctx, clusterName)
 	if err != nil {
 		return err
 	}
-	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+	kubeConfigFlags.Namespace = &cfg.OperatorNamespace
+	kubeConfigFlags.APIServer = &cfg.ServerAPI
+	kubeConfigFlags.BearerToken = &cfg.Token
+	kubeconfig, err := client.EnsureKsctlConfigFile()
 	if err != nil {
 		return err
 	}
+	kubeConfigFlags.KubeConfig = &kubeconfig
+	factory := cmdutil.NewFactory(cmdutil.NewMatchVersionFlags(kubeConfigFlags))
 
-	if len(deployments) == 0 {
-		err := printExistingDeployments(ctx.Terminal, cl, cfg.OperatorNamespace)
-		if err != nil {
-			ctx.Terminal.Printlnf("\nERROR: Failed to list existing deployments\n :%s", err.Error())
-		}
-		return fmt.Errorf("at least one deployment name is required, include one or more of the above deployments to restart")
+	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+	if err != nil {
+		return err
 	}
-	deploymentName := deployments[0]
 
 	if !ctx.AskForConfirmation(
-		ioutils.WithMessagef("restart the deployment '%s' in namespace '%s'", deploymentName, cfg.OperatorNamespace)) {
+		ioutils.WithMessagef("restart all the deployments in the cluster  '%s' and namespace '%s' \n", clusterName, cfg.OperatorNamespace)) {
 		return nil
 	}
-	return restartDeployment(ctx, cl, cfg.OperatorNamespace, deploymentName)
+
+	return restartDeployment(ctx, cl, cfg.OperatorNamespace, factory, ioStreams, checkRolloutStatus, restartNonOperatorDeployments)
 }
 
-func restartDeployment(ctx *clicontext.CommandContext, cl runtimeclient.Client, ns string, deploymentName string) error {
-	namespacedName := types.NamespacedName{
-		Namespace: ns,
-		Name:      deploymentName,
-	}
+// This function has the whole logic of getting the list of operator and non-operator based deployment, then proceed on restarting/deleting accordingly
+func restartDeployment(ctx *clicontext.CommandContext, cl runtimeclient.Client, ns string, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, checker RolloutStatusCheckerFunc, restarter NonOperatorDeploymentsRestarterFunc) error {
```

Comment:
having a function with 7 parameters is a bit hard to read. Also, there is no need to pass the factory and IOStream as a parameter and not having it included in the function itself already from the caller.

Let's change the declaration of the `RolloutStatusCheckerFunc` `NonOperatorDeploymentsRestarterFunc` functions so they don't require the `cmdutil.Factory` and `genericclioptions.IOStreams` to be provided when being called:
```go
type (
	NonOperatorDeploymentsRestarterFunc func(ctx *clicontext.CommandContext, deployment appsv1.Deployment) error
	RolloutStatusCheckerFunc            func(ctx *clicontext.CommandContext, labelSelector string) error
)
```
so the `restartDeployment` can be called with:
```go
return restartDeployment(ctx, cl, cfg.OperatorNamespace, func(ctx *clicontext.CommandContext, labelSelector string) error {
	return checkRolloutStatus(ctx, factory, ioStreams, labelSelector)
}, func(ctx *clicontext.CommandContext, deployment appsv1.Deployment) error {
	return restartNonOperatorDeployments(ctx, deployment, factory, ioStreams)
})
```

Reply:
the function should be also renamed to `restartDeployments` as it restart more deployments than one :-)

Reply:
okay done